### PR TITLE
Recursion and stack

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Line **48** of **"output-single-linked-list"** solution and line **36** of **"output-single-linked-list-reverse"** solution need more attention
Please review and comment/commit if anything is wrong

### CODE REVIEW

File: 1-js/06-advanced-functions/01-recursion/04-output-single-linked-list/solution.md [hunk 1]
```diff
@@ -43,15 +43,15 @@ function printList(list) {
 }
 ```
 
-...But that would be unwise. In the future we may need to extend a function, do something else with the list. If we change `list`, then we lose such ability.
+...اما این کار عاقلانه نیست. در آینده ممکن است تابعی ایجاد کنیم که کار دیگری با لیست انجام می‌دهد. اگر ما `list` را تغییر دهیم، سپس چنین امکانی را از دست می‌دهیم.
 
-Talking about good variable names, `list` here is the list itself. The first element of it. And it should remain like that. That's clear and reliable.
+درباره اسامی خوب متغیر حرف بزنیم، `list` اینجا خودش یک لیست است. اولین المان آن. و ممکن است همانطور بماند. این موضوع واضح و قابل اطمینان است.
```
Comment (mahdyar) [inline 1-js/06-advanced-functions/01-recursion/04-output-single-linked-list/solution.md:48]:
```suggestion
صحبت دربارهٔ اسامی خوب برای متغیرها شد، `list` اینجا خودش یک لیست است. اولین المان آن. و بهتر است همان‌طور بماند. این‌طوری واضح و اتکاپذیر است.
```

File: 1-js/06-advanced-functions/01-recursion/05-output-single-linked-list-reverse/solution.md [hunk 2]
```diff
@@ -31,13 +31,13 @@ function printReverseList(list) {
 printReverseList(list);
 ```
 
-# Using a loop
+# با استفاده از حلقه
 
-The loop variant is also a little bit more complicated then the direct output.
```
Comment (mahdyar) [inline 1-js/06-advanced-functions/01-recursion/05-output-single-linked-list-reverse/solution.md:36]: It's a typo, it should've been like this:
>  The loop variant is also a little bit more complicated *than* the direct output.

Made a PR: https://github.com/javascript-tutorial/en.javascript.info/pull/2731

File: 1-js/06-advanced-functions/01-recursion/05-output-single-linked-list-reverse/solution.md [hunk 3]
```diff
@@ -31,13 +31,13 @@ function printReverseList(list) {
...
+نوع حلقه کمی پیچیده‌تر است سپس به صورت مستقیم نمایش می‌دهیم.
```
Comment (mahdyar) [inline 1-js/06-advanced-functions/01-recursion/05-output-single-linked-list-reverse/solution.md:36]:
```suggestion
روش حلقه کمی پیچیده‌تر از نمایش‌دادن به صورت مستقیم است.
```

File: 1-js/06-advanced-functions/01-recursion/article.md [hunk 4]
```diff
@@ -1,34 +1,34 @@
-# Recursion and stack
+# بازگشت و استک
```
Comment (mahdyar) [inline 1-js/06-advanced-functions/01-recursion/article.md:1]: Haven't you translated the word "stack" to "پشته" in other articles? 🤔

**Review by mahdyar — CHANGES_REQUESTED** [review]

File: 1-js/06-advanced-functions/01-recursion/article.md (hunk 4 above)
Comment (mahdiHashemi14) [reply 1-js/06-advanced-functions/01-recursion/article.md:1]: Did I? To be honest,  I can't remember😅
I'll check it out and change "stack" translation if it's "پشته" in other articles

Comment (mahdiHashemi14) [reply 1-js/06-advanced-functions/01-recursion/article.md:1]: Yes, you're right. I translated it as "پشته". I'll change the article.

**Comment by mahdiHashemi14** [comment]
Thank you @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHashemi14!

**Pull request merged**